src/components/ui/OakTextInput/OakTextInput.stories.tsx

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import { Meta, StoryObj } from "@storybook/react";
+
+import { OakTextInput } from "./OakTextInput";
+
+import { sizeArgTypes } from "@/storybook-helpers/sizeStyleHelpers";
+
+const meta: Meta<typeof OakTextInput> = {
+  component: OakTextInput,
+  tags: ["autodocs"],
+  title: "components/ui/OakTextInput",
+  argTypes: {
+    $width: sizeArgTypes["$width"],
+    $maxWidth: sizeArgTypes["$maxWidth"],
+  },
+  args: {
+    placeholder: "Placeholder text",
+  },
+};
+export default meta;
+
+type Story = StoryObj<typeof OakTextInput>;
+
+export const Default: Story = {
+  render: (args) => <OakTextInput {...args} />,
+};
+
+export const WithStartEnhancer: Story = {
+  render: (args) => (
+    <OakTextInput
+      {...args}
+      value="A fine text value"
+      startEnhancerIconName="search"
+    />
+  ),
+};
+
+export const WithEndEnhancer: Story = {
+  render: (args) => (
+    <OakTextInput
+      {...args}
+      value="A fine text value"
+      endEnhancerIconName="search"
+    />
+  ),
+};
+
+export const WithStyling: Story = {
+  render: (args) => <OakTextInput {...args} />,
+  args: {
+    value: "a test value",
+    $background: "aqua",
+    $color: "blue",
+    $borderColor: "blue",
+    $hoverBackground: "aqua110",
+    $focusRingDropShadows: ["drop-shadow-wide-yellow"],
+  },
+};
+
+export const Disabled: Story = {
+  render: () => <OakTextInput disabled value="A fine text value" />,
+};
+
+export const ReadOnly: Story = {
+  render: () => <OakTextInput readOnly value="A fine text value" />,
+};
+
+export const Valid: Story = {
+  render: () => <OakTextInput validity="valid" value="A fine text value" />,
+};
+
+export const Invalid: Story = {
+  render: () => <OakTextInput validity="invalid" value="A fine text value" />,
+};
+
+export const ValidWithStartEnhancer: Story = {
+  render: () => (
+    <OakTextInput
+      validity="valid"
+      value="A fine text value"
+      startEnhancerIconName="tick"
```

Comment:
Alt text for icons should generally convey meaning not appearance, so this should be 'correct', I think?

Reply:
Actually this is just the icon name, isn't it? In which case my beef is with `OakIcon`, not this file.

Reply:
OakIcon defaults to the Icon name but you can override to whatever you want.

Reply:
🤔 in this case I think the icons are presentational since they're not strictly content. It's only a quirk of implementation that its not added as a background image or another CSS decoration.

It's probably outside the scope of this component (and more the job of a specific input component geared toward quizzes) but would something like `aria-invalid` with `aria-describedBy` be appropriate? https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA21.html

I'll be moving onto that next I suspect so I might ask for feedback when that comes up.

Reply:
@carlmw In this context it's decorative because you're merely demonstrating the capabilities of the component. However in-situ we need to consider what the meaning/purposes of these icons are - if it's displaying a tick or cross (after the user has submitted their answer to be marked) then the icon is not merely presentational/decorative.

I would argue that `aria-invalid` is the wrong choice here for post-submission feedback, because the input itself was valid, but the answer the user gave was incorrect as per the marking scheme. And as far as I know there's no ARIA property called `aria-valid` or similar.

That being said, I'm happy to park this for now and revisit it when the `OakQuizTextInput` component (or whatever) is created.